scrape/scrape_test.go

```diff
@@ -1928,3 +1970,66 @@ func TestCheckAddError(t *testing.T) {
 	sl.checkAddError(nil, nil, nil, storage.ErrOutOfOrderSample, nil, &appErrs)
 	testutil.Equals(t, 1, appErrs.numOutOfOrder)
 }
+
+func TestScrapeReportIsolation(t *testing.T) {
```

Comment:
I'm not seeing how this is testing isolation, as there's nothing to ensure the query runs between the scrape ingestion and the report ingestion.

Reply:
Queries run continuously. We should always have a multiple of 9 metrics.

That test is failing almost instantly without the code.

Reply:
Yes, I was assuming it would happen as @roidelapluie has confirmed now.

Reply:
The test name is misleading, as this doesn't test anything about isolation per-se. What it's testing is that it's all done in one append.

Reply:
@roidelapluie that name should be easy to fix.
@brian-brazil is that the only remaining concern in this PR?

Reply:
Yes, the rest I can live with.

Reply:
Renamed. I originally named it like this because if the underlying appender does not support isolation, the test failed as well.